core/src/main/java/com/linecorp/armeria/server/misc/Inet6AddressBlock.java

```diff
@@ -73,8 +85,13 @@ private static BigInteger ipv6AddressToBigInteger(InetAddress inetAddress) {
         final byte[] ipv6;
         if (inetAddress instanceof Inet4Address) {
             final byte[] ipv4 = inetAddress.getAddress();
-            // Fill the lower 4 bytes with the IPv4 address.
-            ipv6 = new byte[] { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ipv4[0], ipv4[1], ipv4[2], ipv4[3] };
+            if (ipv4[0] == 0x7F && ipv4[1] == 0x00 && ipv4[2] == 0x00 && ipv4[3] == 0x01) {
```

Comment:
`ipv4[3]` can be any number actually, e.g. `127.0.0.2` and `127.0.0.3` are also localhost and thus needs to be translated into `::1`.

Reply:
Thanks! You're right. :-)

From https://tools.ietf.org/html/rfc5735#section-3:
> 127.0.0.0/8 - This block is assigned for use as the Internet host
   loopback address.  A datagram sent by a higher-level protocol to an
   address anywhere within this block loops back inside the host.  This
   is ordinarily implemented using only 127.0.0.1/32 for loopback.  As
   described in [RFC1122], Section 3.2.1.3, addresses within the entire
   127.0.0.0/8 block do not legitimately appear on any network anywhere.

It seems to be better if we check the first byte only. Please let me know if you have any concern.